File: src/test/unit/languages/javascript/fixtures/re-exports.js
/**
 * Re-export Pattern Test Fixture
 * Tests module re-export patterns (export { X } from './path')
 */

// Re-export specific items from other modules
export { UserService } from './class-based.js';
export { validateUser, createUser } from './functional.js';

// Re-export all from a module
export * from './type-relationships.js';

// Re-export with alias
export { UserService as DefaultUserService } from './class-based.js';

// Import and re-export
import { BaseService } from './class-based.js';
export { BaseService };

// Named exports that will be re-exported
export class ConfigService {
    constructor() {
        this.config = {};
    }
    
    getConfig(key) {
        return this.config[key];
    }
}

export function loadConfig(path) {
    return { loaded: true };
}

// This file should create import/re-export relationships

File: packages/analysis/src/test/unit/languages/python/fixtures/re_exports.py
"""
Re-export Pattern Test Fixture
Tests module re-export patterns (from X import Y)
"""

# Re-export specific items from other modules
from .class_based import UserService, BaseService
from .functional import validate_user, create_user

# Re-export with alias
from .class_based import UserRepository as DefaultUserRepository

# Import and re-export all (not recommended but should be parsed)
# from .type_hints import *

# Named exports that will be re-exported
class ConfigService:
    def __init__(self):
        self.config = {}
    
    def get_config(self, key: str):
        return self.config.get(key)

def load_config(path: str) -> dict:
    """Load configuration from file"""
    return {"loaded": True}

# Re-export list (common Python pattern)
__all__ = [
    'UserService',
    'BaseService',
    'validate_user',
    'create_user',
    'DefaultUserRepository',
    'ConfigService',
    'load_config'
]

# This file should create import/re-export relationships
